### PR TITLE
SERV-537: Added sites display

### PR DESCRIPTION
**Pull request by turegjorup**
#### Link to ticket

Please add a link to the ticket being addressed by this change.

#### Description

Added display of ´sites`

#### Screenshot of the result

Screenshot 2022-05-19 at 12 54 30

#### Checklist

- [ ] My code is covered by test cases.
- [ ] My code passes our test (all our tests).
- [ ] My code passes our static analysis suite.
- [ ] My code passes our continuous integration process.

If your code does not pass all the requirements on the checklist you have to add a comment explaining why this change 
should be exempt from the list.

#### Additional comments or questions

If you have any further comments or questions for the reviewer please add them here.

### CODE REVIEW

File: src/Entity/Site.php [hunk 1]
```diff
@@ -95,11 +103,52 @@ public function removeDomain(Domain $domain): self
...
+    public function getPrimaryDomain(): ?string
+    {
+        return $this->primaryDomain;
+    }
+
+    private function setPrimaryDomain(): self
+    {
+        $domainCount = count($this->domains);
+
+        if ($domainCount >= 1) {
```
Comment (cableman) [inline src/Entity/Site.php:148]: A comment about the logic in selecting primary domain would be nice?

File: src/Entity/Site.php [hunk 2]
```diff
@@ -95,11 +103,52 @@ public function removeDomain(Domain $domain): self
...
+
+        return $this;
+    }
+
+    public function getPrimaryDomain(): ?string
+    {
+        return $this->primaryDomain;
+    }
+
+    private function setPrimaryDomain(): self
```
Comment (cableman) [inline src/Entity/Site.php]: Set method that do not take parameters? That requires some documentation?

File: src/Handler/NginxHandler.php [hunk 3]
```diff
@@ -32,6 +38,11 @@ public function handleResult(DetectionResult $detectionResult): void
         try {
             $data = \json_decode($detectionResult->getData(), false, 512, JSON_THROW_ON_ERROR);
 
+            // Nginx 'default' sites should not be indexed.
+            if (str_ends_with($data->config, self::NGINX_DEFAULT)) {
```
Comment (cableman) [inline src/Handler/NginxHandler.php:41]: Are we should that the default domain never is used to serv anything?

File: src/Repository/SiteRepository.php [hunk 4]
```diff
@@ -19,6 +20,18 @@ public function __construct(ManagerRegistry $registry)
         parent::__construct($registry, Site::class);
     }
 
+    public function findByRootDirAndServer(string $rootDir, Server $server)
```
Comment (cableman) [inline src/Repository/SiteRepository.php]: Return type would be nice here?

**Review by cableman — CHANGES_REQUESTED** [review]

File: src/Entity/Site.php (hunk 1 above)
Comment (turegjorup) [reply src/Entity/Site.php:148]: Added

File: src/Entity/Site.php (hunk 2 above)
Comment (turegjorup) [reply src/Entity/Site.php]: Renamed

File: src/Handler/NginxHandler.php (hunk 3 above)
Comment (turegjorup) [reply src/Handler/NginxHandler.php:41]: No. But I haven't seen any examples so far.

File: src/Repository/SiteRepository.php (hunk 4 above)
Comment (turegjorup) [reply src/Repository/SiteRepository.php]: Added. But `getResult()` returns `mixed` so not very usefull

**Review by cableman — APPROVED** [review]

**Pull request merged**